app/components/create-vhost/component.js

```diff
@@ -0,0 +1,40 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+  model: null,
+  services: null,
+  certificates: null,
+  vhostService: null,
+  hasDefaultVhost: false,
+
+  setInitialVhostServiceState: function() {
+    this.setHasDefaultVhost(this.get('vhostService'));
+  }.on('didInsertElement'),
+
+  vhostServiceObserver: function() {
+    this.setHasDefaultVhost(this.get('vhostService'));
+  }.observes('vhostService'),
+
+  setHasDefaultVhost: function(vhostService) {
+    if(vhostService) {
+      vhostService.get('vhosts').then((vhosts) => {
+        let filtered = vhosts.filter((vhost) => vhost.get('isDefault'));
+        this.set('hasDefaultVhost', filtered.length > 0);
+      });
+    }
+  },
```

Comment:
I think you can collapse `setInitalVhostServiceState`, `vhostServiceObserver`, and `setHasDefautlVhost` into a single computed property:


```javascript

 hasDefaultVhost: Ember.computed('vhostService', 'vhostService.vhosts.@each.isDefault', function() {
  let vhostService = this.get('vhostService');
  let hasDefault = false;
  if (vhostService && vhostService.get('vhosts').length > 0) {
    hasDefault = vhostService.get('vhosts').filter((vhost) => vhost.get('isDefault')).length > 0
  }
  return hasDefault;
 })
```